src/components/Card/Card.component.jsx

```diff
@@ -81,47 +82,63 @@ const Icon = styled.div`
 
 function Card({ item }) {
     const [themeContext, themeDispatcher] = useContext(ThemeContext)
+    const [favoriteContext, favoriteDispatcher] = useContext(FavoriteContext)
     const capitalizeFirstLetterSentence = (mySentence) => {
         return mySentence.replace(/(^\w{1})|(\s+\w{1})/g, (letter) => letter.toUpperCase())
     }
+
     return (
         <Container role="gridcell">
-            <img src={item.thumbnails.medium.url} alt={item.title} />
+            <img src={item.snippet.thumbnails.medium.url} alt={item.snippet.title} />
             <Icon>
                 <AiFillEye />
-                <AiFillHeart />
+                <AiFillHeart
+                    onClick={(e) => {
+                        favoriteDispatcher({
+                            type: 'add',
+                            data: item,
+                        })
```

Comment:
I recommend creating functions inside the different providers that include the `dispatch` so that in the components like this one you don't need to use `dispatch` but rather these new functions.  
For example:

in this case instead of using `favoriteDispatcher` you would create a new function `addFunction` inside `FavoriteProvider` that takes in the `item` uses the `dispatcher` internally, and there instead of exposing `favoriteDispatcher` you would expose `addFunction`.  This will make it so that whenever you are using these actions such as on these `onClick` you don't need to use dispatch, remember the type and send the argument as an object.

Reply:
Thank a lot, I'm going to change that part, and about the login, I think it's the credentials problem, I'm going to check it